### DISCUSSION TITLE
never gonna give you up -

### DISCUSSION THREAD

**Discussion by wdiechmann** [yippee-fun/phlex#837] [Help]
I know what it sounds like - but no, I'm not rick-rolling you 🤣 

Anticipating 2.0 I've decided to try lining up early to the party gettin' a feel for the amount of work awaiting me in order to rewrite my app for 2.0 - but I don't know, it's like this `@buffer` never sinks in with me (remember the [form_with and friends](https://github.com/orgs/phlex-ruby/discussions/503) issue of "Christmas Past") so what do you say - got one more for _the old Man_?

Say I'd like to "write some text and part of it will be **strong** and some even **STRONGER**" (it's a lot more complex and ambitious than that but I reckon this will suffice to get me 'cross)

Reading the new https://beta.phlex.fun/ from `doctype` to `</html>` I figured I could do something like

```
<%= render Components::Text.new do |t|
    span { "write some text and part of it will be" }
    strong { "strong" }
    span { "and some even" }
    Components:TextStrong.new() { "STRONGER" } 
end %>

```

using

```
class Components::Text < Components::Base
  attr_accessor :css

  def initialize(css: nil)
    @css = css || "text-base/6 text-zinc-500 sm:text-sm/6 dark:text-zinc-400"
  end

  def view_template(&)
    p(class: css) { yield }
  end
end

# frozen_string_literal: true

class Components::TextStrong < Components::Base
  attr_accessor :css

  def initialize(css: nil)
    @css = css || "font-medium text-zinc-950 dark:text-white"
  end

  def view_template(&)
    strong(class: css) { yield }
  end
end


```
\- but ah-no! 

If I do 

```
<%= render Components::Text.new do |c|
      c.span {"write some text and part of it will be"}
      c.strong {"strong"}
    end %>
```

I can have 'strong' (as a small token of appreciation for all my _elephantizing-the-glass-store_

So - there's the lay of the land - but you gotta' give me some credit for sticking with Phlex for all this time 
😅 not knowing what the f* I'm doing 😎

**Comment by joeldrapper** [comment]
When rendering Phlex from ERB, you should use it more like this.

```erb
<%= render Components::Text.new do %>
  <span>write some text and part of it will be</span>
  <strong>strong</strong>
  <span>and some even</span>
  <%= render Components:TextStrong.new do %>
    STRONGER
  <% end %>
<% end %>
```

The block passed in will be treated by Phlex as if it was an ERB block, not a Phlex block.

**Comment by wdiechmann** [comment]
I erred when digesting https://beta.phlex.fun/handbook/under-the-hood.html effectively reasoning the @buffer to be "carried forward" in a way that would allow me to keep adding to the buffer from component, its 'children', ad infinitum - my apologies

That, however, begs the question: how does one structure a "dynamic" set of components, perhaps eval'ed at runtime - read: action_text kind'a input with markdown or liquid of sorts 🤔
